### PR TITLE
Adding take-last

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Adding `take-last` phel function

## 🔖 Changes

- Added a new phel function `take-last`

### CODE REVIEW

File: src/phel/core/sequence-functions.phel [hunk 1]
```diff
@@ -96,6 +96,11 @@ collection in map"))
   [n xs]
   (slice xs 0 (if (php/< n 0) 0 n)))
 
+(defn take-last
+  "Takes the last `n` elements of `xs`."
+  [n xs]
+  (if (php/<= n 0) @[] (slice xs (php/* -1 n))))
```
Comment (Chemaclass) [inline src/phel/core/sequence-functions.phel:102]: I am not using `(- n)` (and instead `php/* -1`) because the math module where the function `-` is defined is not loaded yet. Here we are getting into the dependency order problem, where we are trying to use core functions that aren't loaded yet ... I am talking about these loaders inside the `core.phel`. We need to think about this problem and how to solve it in the future 🤔 
```lisp
(load (str __DIR__ "/core/basic-constructors.phel"))
(load (str __DIR__ "/core/basic-sequence-operation.phel"))
(load (str __DIR__ "/core/control-structures.phel"))
(load (str __DIR__ "/core/boolean-operation.phel"))
(load (str __DIR__ "/core/type-operation.phel"))
(load (str __DIR__ "/core/sequence-operation.phel"))
(load (str __DIR__ "/core/for-loop.phel"))
(load (str __DIR__ "/core/sequence-functions.phel"))
(load (str __DIR__ "/core/set-operation.phel"))
(load (str __DIR__ "/core/function-operation.phel"))
(load (str __DIR__ "/core/more-sequence-operation.phel"))
(load (str __DIR__ "/core/bitwise-operation.phel"))
(load (str __DIR__ "/core/math-operation.phel"))
(load (str __DIR__ "/core/print-operation.phel"))
(load (str __DIR__ "/core/threading-macros.phel"))
```

Comment (jenshaase) [reply src/phel/core/sequence-functions.phel:102]: That is ok for me.

> We need to think about this problem and how to solve it in the future

What do you mean exactly? Phels code is compiled/executed in order. Therefore, you can only use function that are already defined before.

File: tests/phel/test/core/sequence-functions.phel [hunk 2]
```diff
@@ -61,7 +61,21 @@
...
+  (is (= @["c"] (take-last 1 @["a" "b" "c"])) "take-last element")
+  (is (= @["b" "c"] (take-last 2 @["a" "b" "c"])) "take-last two elements")
+  (is (= @["a" "b" "c"] (take-last 3 @["a" "b" "c"])) "take-last three elements")
+  (is (= @["a" "b" "c"] (take-last 4 @["a" "b" "c"])) "take-last four elements")
+  (is (= @[] (take-last 0 @["a" "b" "c"])) "take-last zero elements")
+  (is (= @[] (take-last -1 @["a" "b" "c"])) "take-last with negative number")
+  (is (= ["b" "c"] (take-last 2 ["a" "b" "c"])) "take-last on tuple")
+  (is (= @[] (take-last -1 ["a" "b" "c"])) "take-last with negative number on tuple")
+  (is (= (php/array "b" "c") (take-last 2 (php/array "a" "b" "c"))) "take-last on php array")
+  (is (= @[] (take-last -1 (php/array "a" "b" "c"))) "take-last with negative number on php array"))
```
[the remark this answers is not in the document]
Comment (jenshaase) [reply tests/phel/test/core/sequence-functions.phel:78]: I don't have a good solution for it. For me this is ok.

[the remark this answers is not in the document]
Comment (jenshaase) [reply tests/phel/test/core/sequence-functions.phel:78]: Could you add some test for an empty array?

File: src/phel/core/sequence-functions.phel (hunk 1 above)
Comment (Chemaclass) [reply src/phel/core/sequence-functions.phel:102]: I mean, we need to be aware of this, that depending on the order of the functions in the core we can use them or not. This was a perfect example where we cannot use `(- n)` and instead we have to use `(php/* 1)`. Anyway, not a big deal, and yes, it's in the nature of the lang, so there is no solution for this right now. And I think is also fine (to be aware at least)

Comment (jenshaase) [reply src/phel/core/sequence-functions.phel:102]: Ok. I also had the problem after we split the file. When it was in one file it was much easier to reason about the order.

File: tests/phel/test/core/sequence-functions.phel (hunk 2 above)
[the remark this answers is not in the document]
Comment (Chemaclass) [reply tests/phel/test/core/sequence-functions.phel:80]: done

**Review by jenshaase — APPROVED** [review]

**Comment by jenshaase** [comment]
Thanks. I merged it!
